model/load_balancer.rb

```diff
@@ -21,14 +22,24 @@ def hyper_tag_name(project)
   def add_vm(vm)
     DB.transaction do
       incr_update_load_balancer
+      incr_rewrite_dns_records
       super
     end
   end
 
   def detach_vm(vm)
     DB.transaction do
-      DB[:load_balancers_vms].where(load_balancer_id: id, vm_id: vm.id).delete(force: true)
+      DB[:load_balancers_vms].where(load_balancer_id: id, vm_id: vm.id).update(state: "evacuating")
       incr_update_load_balancer
+      incr_rewrite_dns_records
     end
   end
+
+  def remove_vm(vm)
+    DB[:load_balancers_vms].where(load_balancer_id: id, vm_id: vm.id).delete(force: true)
```

Comment:
....but here is delete again, skipping auditing.

Reply:
Audit still can be done through the archived health probe record. Destroy doesn't work here because I did not add a model for load_balancers_vms.

Reply:
Okay, I see, it's a pure join table, with no model....hmm that's a little weird, but I can't deny (some formats of) economy here. I kinda wonder if it's going to lead you to a weird path dependency where people don't put stuff on that model that "ought" to exist because they don't feel like introducing it. 

I can go either way, I guess were it me, I'd make a model.